extension/awsmiddleware/middleware_test.go

```diff
@@ -161,27 +169,39 @@ func TestAppendOrder(t *testing.T) {
 	}
 	for name, testCase := range testCases {
 		t.Run(name, func(t *testing.T) {
-			middleware := &testMiddlewareExtension{}
 			recorder := &recordOrder{}
+			var requestHandlers []RequestHandler
 			for _, handler := range testCase.requestHandlers {
-				handler.handleRequest = recorder.handle(handler.id)
-				middleware.requestHandlers = append(middleware.requestHandlers, handler)
+				handler.handleRequest = recorder.Handle(handler.id)
+				requestHandlers = append(requestHandlers, handler)
 			}
+			handler := new(MockHandler)
+			handler.On("ID").Return("mock")
+			handler.On("Position").Return(After)
+			handler.On("HandleRequest", mock.Anything, mock.Anything)
+			handler.On("HandleResponse", mock.Anything, mock.Anything)
+			requestHandlers = append(requestHandlers, handler)
```

Comment:
Why is `requestHandlers = append(requestHandlers, handler)` done inside the loop on line 176 and again here on line 183?

Reply:
They're adding different handlers to the same slice.